src/partialRightBindTo.ts

```diff
@@ -0,0 +1,74 @@
+import { purry } from './purry';
+
+type RemoveSuffix<
+  P extends ReadonlyArray<unknown>,
+  U extends ReadonlyArray<unknown>,
+> = P extends readonly [...infer T, ...U]
+  ? T
```

Comment:
Similar comment to the above, the naming here is confusing to me, Usually T should be reserved to the "main" actor in the type, and the others could be named more descriptivly.
```suggestion
type RemoveSuffix<
  T extends IterableContainer,
  Suffix extends IterableContainer,
> = T extends readonly [...infer Rest, ...Suffix]
  ? Rest
```